### PR TITLE
Add requests-kerberos

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR adds a recipe for [`requests-kerberos`](https://pypi.org/project/requests-kerberos/).

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
@conda-forge/staged-recipes this is ready, thanks.

File: recipes/requests-kerberos/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "9d21f15241c53c2ad47e813138b9aee4b9acdd04b82048c4388ade15f40a52fd"
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+
```
Comment (npavlovikj) [inline recipes/requests-kerberos/meta.yaml]: Please add `'noarch: python`.

File: recipes/requests-kerberos/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "9d21f15241c53c2ad47e813138b9aee4b9acdd04b82048c4388ade15f40a52fd"
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
```
Comment (npavlovikj) [inline recipes/requests-kerberos/meta.yaml]: You can just use `"{{ PYTHON }} -m pip install . -vv"` here (it actually does --no-deps --ignore-installed).

File: recipes/requests-kerberos/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+requirements:
+  host:
+    - pip
+    - python
+    - setuptools
```
Comment (npavlovikj) [inline recipes/requests-kerberos/meta.yaml]: Is `setuptools` really needed here? Usually `pip` and `python` are enough for the `host` section.

File: recipes/requests-kerberos/meta.yaml (hunk 1 above)
Comment (duncanmmacleod) [reply recipes/requests-kerberos/meta.yaml]: @npavlovikj, my understanding is that `noarch: python` is not appropriate for packages that have platform-specific runtime requirements, as this does.

File: recipes/requests-kerberos/meta.yaml (hunk 2 above)
Comment (duncanmmacleod) [reply recipes/requests-kerberos/meta.yaml]: This is just what I get from `conda skeleton pypi`, but I can change it if requested.

File: recipes/requests-kerberos/meta.yaml (hunk 3 above)
Comment (duncanmmacleod) [reply recipes/requests-kerberos/meta.yaml]: `setuptools` is required for the build of this package, so I would rather list is explicitly than rely upon the fact that `pip` happens to also depend upon `setuptools`.

**Comment by npavlovikj** [comment]
Thank you @duncanmmacleod 🎉

**Pull request merged**